main.py

```diff
@@ -0,0 +1,59 @@
+import sys
+
+import streamlit as st
+import torch
+from diffusers import StableDiffusionPipeline
+import requests
+from bs4 import BeautifulSoup
+from googletrans import Translator
+def check_url(url):
+    r = requests.head(url)
+    return r.status_code == 200
+
+def get_text_from_site(url):
+    try:
+        response = requests.get(url)
+        bs = BeautifulSoup(response.text, 'html.parser')
+        part_of_text = bs.find('div', 'c-detail text').text
+        index = part_of_text.lower().find("приметы")
+        i = part_of_text.find(":", index)
+        result_sentence = part_of_text[i+1:part_of_text.find("\n", i+1)].replace(".", "")
+        translator = Translator()
+        translation = translator.translate(result_sentence, src='ru', dest='en')
+        result_sentence = translation.text
+    except:
+        result_sentence = ""
+    return result_sentence
+
+def generate_arcane(text):
+    model_id = "nitrosocke/Arcane-Diffusion"
+    if torch.cuda.is_available():
+        pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16)
+        pipe = pipe.to("cuda")
+    else:
+        pipe = StableDiffusionPipeline.from_pretrained(model_id)
+        pipe = pipe.to("cpu")
```

Comment:
First of all, with this behavior, you will load the model for every generation. It is both time and memory-consuming. It is better to load the pipeline only once on app start and then reuse the pipeline.

Secondly, you can write it a little bit prettier 
```python
device = "cuda" if torch.cuda.is_available() else "cpu"
pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16).to(device)
```

Reply:
It won't work. torch_dtype=torch.float16 and cpu are incompatible.

Reply:
How to load it once if streamlit does a permanent reload?

Reply:
> torch_dtype=torch.float16 and cpu are incompatible.
Yeah, we figure it out in another pull request. You still can optimize this code a little bit. There is no need for `.to("cpu")` model will load on it by default. And also add `revision="fp16"` in the loading of the optimized version.

> How to load it once if streamlit does a permanent reload?
Well, you need to separate UI and model. For example, you can create a standalone class that load all models and handle text input. After that, you can pass this class to streamlit interface builder.